libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -449,6 +460,70 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CWriteThenReadAsyncSuccess )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Test Function for I2C async read success
+ */
+TEST( TEST_IOT_I2C, AFQP_IotI2CReadAsyncAssisted )
+{
+    IotI2CHandle_t xI2CHandle;
+    int32_t lRetVal;
+    uint8_t ucReadBuf[16];
+    char cMsg[testIotI2C_MESSAGE_LENGTH] = {0};
+    size_t msgOffset = 0;
+
+    IotI2CConfig_t xI2CConfig = {
+        .ulBusFreq = testIotI2C_BAUDRATE,
+        .ulMasterTimeout =  testIotI2C_DEFAULT_TIMEOUT
+    };
+
+    if(TEST_PROTECT())
+    {
+        /* Open i2c to initialize hardware */
+        xI2CHandle = iot_i2c_open(testIotI2C_INSTANCE);
```

Comment:
iot_i2c_open needs to happen before TEST_PROTECT(). Otherwise, even if open fails, iot_i2c_close will also be called.

Reply:
done